src/lib/webSocketClient.ts

```diff
@@ -0,0 +1,138 @@
+import { v4 as genUuid } from 'uuid';
+import camelcaseKeysDeep from 'camelcase-keys-deep';
+import { IMiddlewareWebSocketSubscriptionMessage, WebSocketChannelName } from '../types';
+import {
+  WEB_SOCKET_CHANNEL_KEY_BLOCKS,
+  WEB_SOCKET_CHANNEL_KEY_OBJECT,
+  WEB_SOCKET_CHANNEL_MICRO_BLOCKS,
+  WEB_SOCKET_CHANNEL_TRANSACTIONS,
+  WEB_SOCKET_SUBSCRIBE,
+  handleUnknownError,
+} from '../popup/utils';
+
+let wsClient: WebSocket;
+let isWsConnected: boolean;
+
+const subscribersQueue: IMiddlewareWebSocketSubscriptionMessage[] = [];
+// eslint-disable-next-line no-unused-vars
+const subscribers:Record<WebSocketChannelName, Record<string, (payload: any) => void>> = {
+  [WEB_SOCKET_CHANNEL_TRANSACTIONS]: {},
+  [WEB_SOCKET_CHANNEL_MICRO_BLOCKS]: {},
+  [WEB_SOCKET_CHANNEL_KEY_BLOCKS]: {},
+  [WEB_SOCKET_CHANNEL_KEY_OBJECT]: {},
+};
+
+export default class WebSocketClient {
+  static handleWebsocketOpen() {
+    isWsConnected = true;
+    subscribersQueue.forEach((message) => {
+      wsClient.send(JSON.stringify(message));
+    });
+  }
+
+  static handleWebsocketClose() {
+    isWsConnected = false;
+  }
+
+  static isWsConnected() {
+    return isWsConnected;
+  }
+
+  static subscribeForChannel(
+    message: IMiddlewareWebSocketSubscriptionMessage,
+    // eslint-disable-next-line no-unused-vars
+    callback: (payload: any) => void,
+  ) {
+    if (isWsConnected) {
+      wsClient.send(JSON.stringify(message));
+    }
+
+    subscribersQueue.push(message);
```

Comment:
If web socket is connected then there is no need to add the new subscription to the cache queue.
```suggestion
    if (isWsConnected) {
      wsClient.send(JSON.stringify(message));
    } else {
      subscribersQueue.push(message);
    }
```

Reply:
we need to keep the list of subscribers for network change event